### PR TITLE
[snx-multichain] feat: Add synthetix collateral voting strategy

### PR DESCRIPTION
**Pull request by barrasso**
Changes proposed in this pull request:
- Added support for fetching voting power from Synthetix V2 + V3 by querying `SNX` collateral across all supported networks.
- Added logic to enumerate account IDs and fetch collateral values using `AccountProxy` and `CoreProxy` contracts.
- Updated `examples.json` to reflect the correct configurations for v2x and v3 strategies on supported networks.

### CODE REVIEW

**Review by ChaituVR — COMMENTED** [review]
Looks good other than above :)

File: src/strategies/snx-multichain/index.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            accountId,
+            options.collateralType
+          );
+
+        // Sum the collateral values
+        totalCollateral += totalDeposited + totalAssigned + totalLocked;
+      }
+    } catch (err) {
+      // Exit the loop when all tokens are enumerated
+    }
```
[the remark this answers is not in the document]
Comment (barrasso) [reply src/strategies/snx-multichain/index.ts]: Updated 3c781a47d78e2b7e4e2b090fe043b8c773b68447

File: src/strategies/snx-multichain/index.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const promises: any = [];
+  const validStrategies = options.strategies.filter(
+    (s) => s.network === '1' || s.network === '10' || s.network === '8453'
+  );
```
[the remark this answers is not in the document]
Comment (barrasso) [reply src/strategies/snx-multichain/index.ts:48]: Updated 3c781a47d78e2b7e4e2b090fe043b8c773b68447

**Review by ChaituVR — APPROVED** [review]

File: src/strategies/snx-multichain/index.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      provider
+    );
+
+    let totalCollateral = 0;
+
+    try {
+      // Check the number of tokens owned by the address
+      const balance = await accountProxy.balanceOf(address);
+      if (balance.eq(0)) {
+        console.log(`Address ${address} has no tokens.`);
```
Comment (ChaituVR) [inline src/strategies/snx-multichain/index.ts:104]:
```suggestion
```

File: src/strategies/snx-multichain/index.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      // Fetch only the totalDeposited value for the account ID
+      const { totalDeposited } = await coreProxy.getAccountCollateral(
+        accountId,
+        collateralType
+      );
+
+      totalCollateral = totalDeposited;
+    } catch (err) {
+      // Assume the account doesn't exist and continue.
+      console.log(`Error fetching v3 collateral for address ${address}:`, err);
```
Comment (ChaituVR) [inline src/strategies/snx-multichain/index.ts:122]:
```suggestion
      // console.log(`Error fetching v3 collateral for address ${address}:`, err);
```

**Pull request merged**